pkg/distributor/distributor.go

```diff
@@ -100,6 +100,8 @@ type Distributor struct {
 	ingesterPool  *ring_client.Pool
 	limits        *validation.Overrides
 
+	isServiceStarted *atomic.Bool
```

Comment:
We don't need this. We can do `d.State() == services.Running` check instead, similar to how ingesters do it: https://github.com/grafana/mimir/blob/c715b0a90bee797adb8c762e3c27405f44dbefd3/pkg/ingester/ingester.go#L3818-L3824

`d.State()` will switch to `Running` after `starting` method finishes successfully.

Reply:
Thanks, done in https://github.com/grafana/mimir/pull/9317/commits/aecfd7321693ace25b39a595ba80fb3a62d4fd41